src/ServiceControl/Infrastructure/MetadataExtensions.cs

```diff
@@ -24,5 +25,16 @@ public static string GetAsStringOrNull(this IDictionary<string, object> metadata
 
             return null;
         }
+
+        public static DateTime? GetAsNullableDatetime(this IDictionary<string, object> metadata, string key)
+        {
+            var datetimeAsString = metadata.GetAsStringOrNull(key);
+            DateTime dt;
+            if (datetimeAsString != null && DateTime.TryParse(datetimeAsString, out dt))
```

Comment:
Why TryParse? If the format is wrong should we then really import it? If we do import, we should log a WARN that this message was imported but that timesent was skipped.